### PR TITLE
Warn of unsaved changes

### PR DESCRIPTION
**Pull request by afranco07**
Closes https://github.com/bitwarden/desktop/issues/259

Ran into this a couple times and wanted to fix it 😄 should work for both new and edited ones!

### CODE REVIEW

File: src/app/vault/vault.component.ts [hunk 1]
```diff
@@ -223,9 +223,18 @@ export class VaultComponent implements OnInit, OnDestroy {
         });
     }
 
-    viewCipher(cipher: CipherView) {
+    async viewCipher(cipher: CipherView) {
         if (this.action === 'view' && this.cipherId === cipher.id) {
             return;
+        } else if (document.getElementsByClassName("ng-dirty").length > 0 && (this.action == 'add' || this.action === 'edit')) {
```
Comment (kspearrin) [inline src/app/vault/vault.component.ts]: What if an input outside of the add/edit forms is dirty? Like the search box?

Comment (afranco07) [reply src/app/vault/vault.component.ts]: @kspearrin good catch! I'm not too familiar with angular, is there a way to check the `AddEdit` components inputs/fields only?

Comment (kspearrin) [reply src/app/vault/vault.component.ts]: You could just limit your `document` query to a section of the DOM tree only applicable to the add/edit pages.

Comment (afranco07) [reply src/app/vault/vault.component.ts]: @kspearrin  Changed it to `document.getElementById('details').getElementsByClassName('ng-dirty')`

**Pull request merged**